File: data/code_pairs/fortran/1221.f90
SUBROUTINE FRPLOT(N, S, X, HK, TH, UE, VE, XTR, FREF, XOFF, XSF, YOFF, YSF, CHF)
  IMPLICIT NONE
  INTEGER, INTENT(INOUT) :: N
  REAL, DIMENSION(N), INTENT(INOUT) :: S, X, HK, TH, UE, VE
  REAL, INTENT(INOUT) :: XTR, FREF, XOFF, XSF, YOFF, YSF
  CHARACTER, INTENT(INOUT) :: CHF
  
  N = 0
END SUBROUTINE FRPLOT

PROGRAM testFRPLOT
  IMPLICIT NONE
  INTEGER :: N
  REAL, ALLOCATABLE :: S(:), X(:), HK(:), TH(:), UE(:), VE(:)
  REAL :: XTR, FREF, XOFF, XSF, YOFF, YSF
  CHARACTER :: CHF
  
  N = 10
  ALLOCATE(S(N), X(N), HK(N), TH(N), UE(N), VE(N))
  
  CALL FRPLOT(N, S, X, HK, TH, UE, VE, XTR, FREF, XOFF, XSF, YOFF, YSF, CHF)
  
  IF (N == 0) THEN
    PRINT *, "Test passed: N is set to 0."
  ELSE
    PRINT *, "Test failed: N is not set to 0."
  END IF
END PROGRAM testFRPLOT

File: data/code_pairs/cpp/1221.cpp
#include <iostream>
#include <vector>
#include <cassert>

void FRPLOT(int& N, std::vector<double>& S, std::vector<double>& X,
            std::vector<double>& HK, std::vector<double>& TH,
            std::vector<double>& UE, std::vector<double>& VE,
            double& XTR, double& FREF, double& XOFF, double& XSF,
            double& YOFF, double& YSF, char& CHF) {
    N = 0;
}

void testFRPLOT() {
    int N = 10;
    std::vector<double> S(N), X(N), HK(N), TH(N), UE(N), VE(N);
    double XTR, FREF, XOFF, XSF, YOFF, YSF;
    char CHF = 'A';

    FRPLOT(N, S, X, HK, TH, UE, VE, XTR, FREF, XOFF, XSF, YOFF, YSF, CHF);

    assert(N == 0);
    std::cout << "Test passed: N is set to 0." << std::endl;
}

int main() {
    testFRPLOT();
    return 0;
}